geoutils/georaster.py

```diff
@@ -1857,3 +1859,60 @@ def to_points(
             )
 
         return points
+
+    def polygonize(
+        self, in_value: Number | tuple[Number, Number] | list[Number] | np.ndarray | Literal["all"] = 1
+    ) -> Vector:
+        """
+        Return a GeoDataFrame polygonized from a raster.
+
+        :param in_value: Value or range of values of the raster from which to
+          create geometries (Default is 1). If 'all', all unique pixel values of the raster are used.
+
+        :returns: Vector containing the polygonized geometries.
+        """
+
+        # mask a unique value set by a number
+        if isinstance(in_value, Number):
+
+            if np.sum(self.data == in_value) == 0:
+                raise ValueError(f"no pixel with in_value {in_value}")
+
+            bool_msk = np.array(self.data == in_value).astype(np.uint8)
+
+        # mask values within boundaries set by a tuple
+        elif isinstance(in_value, tuple):
+
+            if np.sum((self.data > in_value[0]) & (self.data < in_value[1])) == 0:
+                raise ValueError(f"no pixel with in_value between {in_value[0]} and {in_value[1]}")
+
+            bool_msk = ((self.data > in_value[0]) & (self.data < in_value[1])).astype(np.uint8)
+
+        # mask specific values set by a sequence
+        elif isinstance(in_value, list) or isinstance(in_value, np.ndarray):
+
+            if np.sum(np.isin(self.data, in_value)) == 0:
+                raise ValueError("no pixel with in_value " + ", ".join(map("{}".format, in_value)))
+
+            bool_msk = np.isin(self.data, in_value).astype("uint8")
+
+        # mask all valid values
+        elif in_value == "all":
+
+            vals_for_msk = list(set(self.data.flatten()))
+            bool_msk = np.isin(self.data, vals_for_msk).astype("uint8")
+
+        else:
+
+            raise ValueError("in_value must be a number, a tuple or a sequence")
+
+        results = (
+            {"properties": {"raster_value": v}, "geometry": s}
+            for i, (s, v) in enumerate(shapes(self.data, mask=bool_msk))
```

Comment:
I cloned your branch to try to figure out why the test failed, and it's because this does not include any georeferencing! The outgoing coordinates here are in pixel units, not georeferenced units!

Reply:
Good thing for tests, huh!? ;)

Reply:
I assume that it's as simple as adding `transform=self.transform` to the `shapes(...)` call.

Source: https://rasterio.readthedocs.io/en/latest/api/rasterio.features.html#rasterio.features.shapes